### PR TITLE
FIX DebugViewFrendlyErrorFormatter handle of admin_email

### PR DESCRIPTION
**Pull request by dnsl48**
fixes https://github.com/silverstripe/silverstripe-framework/issues/9251

### CODE REVIEW

File: src/Control/Email/Email.php [hunk 1]
```diff
@@ -49,9 +49,10 @@ class Email extends ViewableData
 
     /**
      * This will be set in the config on a site-by-site basis
+     * @see https://docs.silverstripe.org/en/4/developer_guides/email/#administrator-emails
      *
      * @config
-     * @var string The default administrator email address.
+     * @var string|array The default administrator email address or array of [email => name]
```
Comment (robbieaverill) [inline src/Control/Email/Email.php]: The string value is just an email address, do you think the array value should be a list of email addresses? I find it a little odd that we're changing array values to accept email and names

File: src/Logging/DebugViewFriendlyErrorFormatter.php [hunk 2]
```diff
@@ -127,9 +128,22 @@ public function output($statusCode)
...
-            $output .= $renderer->renderParagraph('Contact an administrator: ' . $mailto . '');
+        if ($adminEmail = Email::config()->admin_email) {
+            if (is_array($adminEmail)) {
+                if (count($adminEmail) > 0) {
+                    $email = array_keys($adminEmail)[0];
+                    $name = array_values($adminEmail)[0];
+
+                    $output .= $renderer->renderParagraph(sprintf(
+                        'Contact %s: %s',
+                        Convert::raw2xml($name),
```
Comment (robbieaverill) [inline src/Logging/DebugViewFriendlyErrorFormatter.php]: There's a bit too much nesting here for my liking - any chance of a private method with early returns, like `$this->getAdminEmails()` and `$this->appendContactInformation($emails)`?

File: src/Control/Email/Email.php (hunk 1 above)
Comment (dnsl48) [reply src/Control/Email/Email.php]: Yeah, I agree that API is utterly bad, but that's what Swiftmailer has and what we pass there "as is" and what we've got in the documentation - https://docs.silverstripe.org/en/4/developer_guides/email/#administrator-emails

Comment (dnsl48) [reply src/Control/Email/Email.php]: I reckon we should fix that in the next major release though

File: src/Logging/DebugViewFriendlyErrorFormatter.php (hunk 2 above)
Comment (dnsl48) [reply src/Logging/DebugViewFriendlyErrorFormatter.php]: done

File: src/Logging/DebugViewFriendlyErrorFormatter.php [hunk 3]
```diff
@@ -127,12 +128,36 @@ public function output($statusCode)
...
         }
 
         $output .= $renderer->renderFooter();
         return $output;
     }
+
+    /**
+     * Generate the line with admin contact info
+     *
+     * @return ?string
```
Comment (robbieaverill) [inline src/Logging/DebugViewFriendlyErrorFormatter.php]: I’m not sure this is valid phpdoc, maybe use string|null

File: src/Logging/DebugViewFriendlyErrorFormatter.php [hunk 4]
```diff
@@ -127,12 +128,36 @@ public function output($statusCode)
...
+     */
+    private function addContactAdministratorInfo(): ?string
+    {
+        if (!$adminEmail = Email::config()->admin_email) {
+            return null;
+        }
+
+        if (is_string($adminEmail)) {
+            return 'Contact an administrator: ' . Email::obfuscate($adminEmail);
```
Comment (robbieaverill) [inline src/Logging/DebugViewFriendlyErrorFormatter.php]: Does this need i18n?

Comment (dnsl48) [reply src/Logging/DebugViewFriendlyErrorFormatter.php]: that would be a minor change, I'm currently doing a patch for the bugfix :)

File: src/Logging/DebugViewFriendlyErrorFormatter.php (hunk 3 above)
Comment (dnsl48) [reply src/Logging/DebugViewFriendlyErrorFormatter.php]: done

File: src/Logging/DebugViewFriendlyErrorFormatter.php (hunk 4 above)
Comment (robbieaverill) [reply src/Logging/DebugViewFriendlyErrorFormatter.php]: Alright. I always treat missing localisation as a bug, but if it was already like that then I guess it’s fine for now :-)

File: src/Logging/DebugViewFriendlyErrorFormatter.php [hunk 5]
```diff
@@ -127,12 +128,36 @@ public function output($statusCode)
...
         $output .= $renderer->renderFooter();
         return $output;
     }
+
+    /**
+     * Generate the line with admin contact info
+     *
+     * @return string|null
+     */
+    private function addContactAdministratorInfo(): ?string
```
Comment (robbieaverill) [inline src/Logging/DebugViewFriendlyErrorFormatter.php]: Missed this- if this is a 4.3 patch then we can’t use PHP7 syntax

Comment (dnsl48) [reply src/Logging/DebugViewFriendlyErrorFormatter.php]: oh, nice catch!
fixed

**Pull request merged**